File: iOS/VkAudio/vk/response_models/Audio.swift
//
//  Audio.swift
//  VkAudio
//
//  Created by mac-224 on 29.04.16.
//  Copyright © 2016 y0rrrsh. All rights reserved.
//

import Foundation

class Audio: VkItem, AudioPlayerItem {

    var url: String?
    var artist: String?
    var name: String?
    var duration: Int?
    var ownerId: Int?

    override init(apiResponse: [String:AnyObject]) {
        super.init(apiResponse: apiResponse)

        id = apiResponse["aid"] as? Int
        url = apiResponse["url"] as? String
        artist = apiResponse["artist"] as? String
        name = apiResponse["title"] as? String
        duration = apiResponse["duration"] as? Int
        ownerId = apiResponse["owner_id"] as? Int
    }
}

File: Android/VKAudioPlayer/app/src/main/java/github/y0rrrsh/vkaudioplayer/models/response/Audio.java
package github.y0rrrsh.vkaudioplayer.models.response;

/**
 * @author Artur Yorsh
 */
public class Audio extends VkItem {

    private String title;
    private String artist;
    private int duration;

    public Audio(long id, String artist, String title, int duration) {
        super(id);
        this.artist = artist;
        this.title = title;
        this.duration = duration;
    }

    public String getTitle() {
        return title;
    }

    public String getArtist() {
        return artist;
    }

    public int getDuration() {
        return duration;
    }
}
